src/app/carbon-estimation-table/carbon-estimation-table.component.ts

```diff
@@ -0,0 +1,323 @@
+import { ChangeDetectorRef, Component, computed, input } from '@angular/core';
+import { CarbonEstimation } from '../types/carbon-estimator';
+import {
+  EmissionsColours,
+  EmissionsLabels,
+  placeholderTableData,
+} from '../carbon-estimation/carbon-estimation.constants';
+import { CarbonEstimationUtilService } from '../services/carbon-estimation-util.service';
+import { NumberObject } from '../utils/number-object';
+import { NgClass, NgStyle } from '@angular/common';
+
+export type TableItem = {
+  category: string;
+  emissions: string;
+  colour: ItemColour;
+  display: boolean;
+  positionInSet: number;
+  setSize: number;
+  parent?: string;
+  svg?: string;
+  expanded?: boolean;
+  level: number;
+};
+
+type ItemColour = {
+  svg?: string;
+  background: string;
+};
+
+type ArrowDirection = ArrowDirectionHorizontal | ArrowDirectionVertical;
+
+type ArrowDirectionHorizontal = 'left' | 'right';
+type ArrowDirectionVertical = 'up' | 'down';
+
+@Component({
+  selector: 'carbon-estimation-table',
+  standalone: true,
+  imports: [NgStyle, NgClass],
+  templateUrl: './carbon-estimation-table.component.html',
+})
+export class CarbonEstimationTableComponent {
+  public carbonEstimation = input<CarbonEstimation>();
+
+  public tableData = computed(() => this.getTableData(this.carbonEstimation()));
+
+  constructor(
+    private carbonEstimationUtilService: CarbonEstimationUtilService,
+    private changeDetector: ChangeDetectorRef
+  ) {}
+
+  public toggle(category: string): void {
+    this.tableData().forEach(emission => {
+      if (emission.parent === category) {
+        emission.display = !emission.display;
+      } else if (emission.category === category) {
+        emission.expanded = !emission.expanded;
+      }
+    });
+  }
+
+  public parentRowArrowKeyBoardEvent(
+    keyBoardEvent: Event,
+    direction: ArrowDirectionHorizontal,
+    parent: string,
+    expanded?: boolean
```

Comment:
What was the reason for making this argument optional?

Reply:
I think it was left over from a refactor. Will remove optionality as it is needed.